include/nbl/builtin/hlsl/shapes/spherical_rectangle.hlsl

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SHAPES_SPHERICAL_RECTANGLE_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SHAPES_SPHERICAL_RECTANGLE_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/numbers.hlsl>
+#include <nbl/builtin/hlsl/math/functions.hlsl>
+#include <nbl/builtin/hlsl/math/angle_adding.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace shapes
+{
+
+template<typename Scalar>
+struct SphericalRectangle
+{
+    using scalar_type = Scalar;
+    using vector3_type = vector<Scalar, 3>;
+    using vector4_type = vector<Scalar, 4>;
```

Comment:
you don't need a `vector4_type` here, you can move the alias to the inside of `soildAngle`